### ISSUE TITLE
How does the model work during training? what's happend in model forward in Trainer._step() ?

### ISSUE THREAD

**Issue by GoldenFishes** [facebookresearch/sam2#582]
what's happend in model forward in Trainer._step() ?
`outputs = model(batch)` in Trainer._step() .
This model was init at  `self.model = instantiate(self.model_conf, _convert_="all")` in Trainer._setup_components() .
I don't understand why we can use `model(batch)` to get mask outputs , the model should be init as `SAM2Base(torch.nn.Module)`. There was no any function can support usage of `outputs = model(batch)`.

Where is forward propagation defined which `Trainer` used at `SAM2Base(torch.nn.Module)` ? 
I can't find a function for full sam2 forward propagation on training.

**Comment by GoldenFishes** [comment]
i’m wrong，it use SAM2Train(SAM2Base)
